src/utils.ts

```diff
@@ -2,14 +2,27 @@ import * as WebBrowser from 'expo-web-browser';
 import { Platform } from 'react-native';
 
 import { PRODUCTION_URL, SANDBOX_URL } from './constants';
-import { DisclosureParams, SoyioWidgetParams } from './types';
+import { DisclosureParams, SignatureParams, SoyioWidgetParams } from './types';
+
+type RequestUrlParams = { request: 'disclosure' } & DisclosureParams
+  | { request: 'signature' } & SignatureParams;
+
+function getPath(params: RequestUrlParams) {
+  if (params.request === 'disclosure' && params.disclosureRequestId) {
+    return `disclosures/${params.disclosureRequestId}`;
+  }
+
+  return params.request;
+}
 
 export function getRequestUrl(
   options: SoyioWidgetParams,
-  request: 'disclosure' | 'signature',
+  params: RequestUrlParams,
 ): string {
   const baseUrl = options.developmentUrl || (options.isSandbox ? SANDBOX_URL : PRODUCTION_URL);
-  return `${baseUrl}/${request}`;
+  const path = getPath(params);
+
+  return `${baseUrl}/${path}`;
 }
```

Comment:
Duda, acá por qué separaste como el armado de la url en 2 funciones?
Creo que me da lo mismo si se hace en 2 o en 1, pero es para cachar le pensamiento nomás jajaj

Reply:
Buena, es más que nada por estilo mio de código, la alternativa que pensé fue:

```ts
export function getRequestUrl(
  options: SoyioWidgetParams,
  params: RequestUrlParams,
): string {
  const baseUrl = options.developmentUrl || (options.isSandbox ? SANDBOX_URL : PRODUCTION_URL);
  let path = `/widget/${params.request}`;

  if (params.disclosureRequestId) {
    path = `/widget/disclosures/${params.disclosureRequestId};
  }

  return `${baseUrl}/${path}`;
}
```

Y creo que generalmente prefiero meterlo en funciones porque por el naming se entiende mejor qué estoy haciendo en esas 4 lineas (generando el path dependiendo de los parámetros). Igual es bien subjetivo esto y quizás no vi otra alternativa, ahí feliz que me cuentes si te hace sentido o si quizás estoy overkilling it!